pkg/iterator/iter.go

```diff
@@ -85,6 +87,25 @@ type Args struct {
 	// identified by SnapshotName will be enumerated.
 	PrevSnapshotName string
 
+	// Verify is optional, and if true then the client will copy changed
+	// blocks from SourceDevice to TargetDevice and verify if the final
+	// contents match.
+	Verify bool
+
+	// SourceDevice is optional, and if specified will be used to copy
+	// changed blocks to the TargetDevice.
+	SourceDevice string
+
+	// TargetDevice is optional, and if specified will be used to copy
+	// changed blocks from the SourceDevice.
+	TargetDevice string
+
+	// Source contains the source device file descriptor.
+	Source *os.File
+
+	// Target contains the target device file descriptor.
+	Target *os.File
+
```

Comment:
I recommend not modifying the iterator package but moving the testing logic to the e2e or example logic.